internal/pkg_npm/pkg_npm.bzl

```diff
@@ -167,77 +173,102 @@ def create_package(ctx, deps_sources, nested_packages):
+        # this action runs, so we get the latest version info on each execution.
+        # See https://github.com/bazelbuild/bazel/issues/1054
+        if stamp:
+            inputs.append(ctx.version_file)
+
+        ctx.actions.run(
+            progress_message = "Assembling npm package %s" % package_dir.short_path,
+            mnemonic = "AssembleNpmPackage",
+            executable = ctx.executable._packager,
+            inputs = inputs,
+            outputs = [package_dir],
+            arguments = [args],
+        )
 
     args = ctx.actions.args()
-    args.use_param_file("%s", use_always = True)
-    args.add(package_dir.path)
-    args.add(package_path)
-    args.add_joined([s.path for s in ctx.files.srcs], join_with = ",", omit_if_empty = False)
-    args.add(ctx.bin_dir.path)
-    args.add(ctx.genfiles_dir.path)
-    args.add_joined(filtered_deps_sources, join_with = ",", omit_if_empty = False)
-    args.add_joined([p.path for p in nested_packages], join_with = ",", omit_if_empty = False)
-    args.add(ctx.attr.substitutions)
-    args.add_all([ctx.outputs.pack.path, ctx.outputs.publish.path])
-    args.add(ctx.attr.replace_with_version)
-    args.add(ctx.version_file.path if stamp else "")
-    args.add_joined(ctx.attr.vendor_external, join_with = ",", omit_if_empty = False)
-    args.add("1" if ctx.attr.hide_build_files else "0")
-
-    # require.resolve expects the path to start with the workspace name and not "external"
-    run_npm_template_path = strip_external(ctx.file._run_npm_template.path)
-    args.add(run_npm_template_path)
-
-    inputs = ctx.files.srcs + deps_sources + nested_packages + [ctx.file._run_npm_template]
-
-    # The version_file is an undocumented attribute of the ctx that lets us read the volatile-status.txt file
-    # produced by the --workspace_status_command. That command will be executed whenever
-    # this action runs, so we get the latest version info on each execution.
-    # See https://github.com/bazelbuild/bazel/issues/1054
-    if stamp:
-        inputs.append(ctx.version_file)
+    args.add_all([
+        package_dir.path,
+        ctx.outputs.pack.path,
+        ctx.outputs.publish.path,
+        ctx.file._run_npm_template.path,
+    ])
 
     ctx.actions.run(
-        progress_message = "Assembling npm package %s" % package_dir.short_path,
-        mnemonic = "AssembleNpmPackage",
-        executable = ctx.executable._packager,
-        inputs = inputs,
-        outputs = [package_dir, ctx.outputs.pack, ctx.outputs.publish],
+        progress_message = "Preparing pack & publish scripts",
+        mnemonic = "PrepareNpmScripts",
+        executable = ctx.executable._npm_script_generator,
+        inputs = [ctx.file._run_npm_template, package_dir],
+        outputs = [ctx.outputs.pack, ctx.outputs.publish],
         arguments = [args],
+        # Must be run local (no sandbox) so that the pwd is the actual execroot
```

Comment:
there's a corresponding edit needed in some docstring where it says you must run from the root of the repo to make the paths work

Reply:
Found it 👍